crates/store/re_data_loader/src/le_robot/mod.rs

```diff
@@ -0,0 +1,275 @@
+            re_tracing::profile_scope!("fs::read");
+            std::fs::read(&videopath)?
+            // TODO(gijsd): Look into using anyhow again?
+            // .with_context(|| format!("Failed to read file {videopath:?}"))?
+        };
+
+        Ok(std::borrow::Cow::Owned(contents))
+    }
+}
+
+#[derive(Debug)]
+pub struct LeRobotDatasetMetadata {
+    pub info: LeRobotDatasetInfo,
+    pub episodes: Vec<LeRobotDatasetEpisode>,
+    pub tasks: Vec<LeRobotDatasetTask>,
+}
+
+impl LeRobotDatasetMetadata {
+    pub fn load_from_directory(metadir: impl AsRef<Path>) -> Result<Self, LeRobotError> {
+        let metadir = metadir.as_ref();
+
+        let info = LeRobotDatasetInfo::load_from_file(metadir.join("info.json"))?;
+        let episodes = load_jsonl_file(metadir.join("episodes.jsonl"))?;
+        let tasks = load_jsonl_file(metadir.join("tasks.jsonl"))?;
+
+        Ok(Self {
+            info,
+            episodes,
+            tasks,
+        })
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct LeRobotDatasetInfo {
+    pub robot_type: String,
+    pub codebase_version: String,
+    pub total_episodes: usize,
+    pub total_frames: usize,
+    pub total_tasks: usize,
+    pub total_videos: usize,
+    pub total_chunks: usize,
+    pub chunks_size: usize,
+    pub data_path: String,
+    pub video_path: String,
+    pub fps: usize,
+    pub features: HashMap<String, Feature>,
+}
+
+impl LeRobotDatasetInfo {
+    pub fn load_from_file(filepath: impl AsRef<Path>) -> Result<Self, LeRobotError> {
+        let info_file = File::open(filepath)?;
+        let reader = BufReader::new(info_file);
+
+        serde_json::from_reader(reader).map_err(|err| err.into())
+    }
+
+    pub fn feature(&self, feature_key: &str) -> Option<&Feature> {
+        self.features.get(feature_key)
+    }
+
+    pub fn get_chunk_index(&self, episode_index: usize) -> Result<usize, LeRobotError> {
+        if episode_index > self.total_episodes {
+            return Err(LeRobotError::InvalidEpisodeIndex(episode_index));
+        }
+
+        // chunk indices start at 0
+        let chunk_idx = episode_index / self.chunks_size;
+        if chunk_idx < self.total_chunks {
+            Ok(chunk_idx)
+        } else {
+            Err(LeRobotError::InvalidChunkIndex(chunk_idx))
+        }
+    }
+
+    pub fn episode_data_path(&self, episode_index: usize) -> Result<PathBuf, LeRobotError> {
+        let chunk = self.get_chunk_index(episode_index)?;
+
+        // TODO(gijsd): Need a better way to handle this, as this only supports the default.
+        Ok(self
+            .data_path
+            .replace("{episode_chunk:03d}", &format!("{chunk:03}"))
+            .replace("{episode_index:06d}", &format!("{episode_index:06}"))
+            .into())
+    }
+
+    /// Get the path to a video observation for a specific episode index.
+    pub fn video_path(
+        &self,
+        observation_key: &str,
+        episode_index: usize,
```

Comment:
We have both episode indices and chunk indices; both encoded as `usize`, leaving room for mixing them up. It could be worth considering adding strong newtypes for these:

```rs
pub struct EpisodeIndex(pub usize);

pub struct ChunkIndex(pub usize);
```